src/core/main.cpp

```diff
@@ -1073,11 +1073,11 @@ static void do_global_ctors_helper()
 
     NEW_CTOR(g_buffer_pool_rx_rwqe,
              buffer_pool(BUFFER_POOL_RX, calc_rx_wqe_buff_size(),
-                         (safe_mce_sys().m_ioctl.user_alloc.flags & IOCTL_USER_ALLOC_RX
-                              ? safe_mce_sys().m_ioctl.user_alloc.memalloc
+                         (safe_mce_sys().user_alloc.flags & USER_ALLOC_RX
+                              ? safe_mce_sys().user_alloc.memalloc
                               : nullptr),
-                         (safe_mce_sys().m_ioctl.user_alloc.flags & IOCTL_USER_ALLOC_RX
-                              ? safe_mce_sys().m_ioctl.user_alloc.memfree
+                         (safe_mce_sys().user_alloc.flags & USER_ALLOC_RX
+                              ? safe_mce_sys().user_alloc.memfree
```

Comment:
just pass memalloc and memfree unconditionally here and in tx buffer pool. Make sure it's nullptr in the constructor.